js/app.js

```diff
@@ -231,7 +231,13 @@ function App() {
 				for (let nodeHelper of nodeHelpers) {
 					nodeHelper.setExpressApp(app);
 					nodeHelper.setSocketIO(io);
-					nodePromises.push(nodeHelper.start());
+
+					try {
+						nodePromises.push(nodeHelper.start());
+					} catch (error) {
```

Comment:
This wouldn't catch errors happening if `nodeHelper.start()` returns a promise which fails. I think you need to make sure the result from `start()` is converted to a promise and then use the .catch() on it.

Reply:
The promise would be catched (and can be accessed in the allSettled results value. Added logging for that.

Reply:
Yeah, I meant it wouldn't be caught in that try-catch and logged. Looks good now.